### PR TITLE
Pandas tz normalize bug workaround

### PR DESCRIPTION
**Pull request by mtydykov**
Adds a workaround and a test for the pandas bug that results in an incorrect timestamp, described here: https://github.com/quantopian/qexec/issues/10456

### CODE REVIEW

File: zipline/pipeline/loaders/utils.py [hunk 1]
```diff
@@ -241,7 +241,7 @@ def normalize_timestamp_to_query_time(df,
     # day and truncate to the date
     df.loc[to_roll_forward, ts_field] = (
         dtidx_local_time[to_roll_forward] + datetime.timedelta(days=1)
-    ).normalize().tz_localize(None).tz_localize('utc')  # cast back to utc
+    ).normalize().tz_convert('utc').normalize()  # set to UTC midnight
```
Comment (ssanderson) [inline zipline/pipeline/loaders/utils.py]: This is broken if you're in a timezone with a negative offset from UTC.

File: zipline/pipeline/loaders/utils.py [hunk 2]
```diff
@@ -237,11 +237,15 @@ def normalize_timestamp_to_query_time(df,
         _midnight,
         include_end=False,
     )
-    # for all of the times that are greater than our query time add 1
-    # day and truncate to the date
+    # For all of the times that are greater than our query time add 1
+    # day and truncate to the date.
+    # We normalize twice here because of a bug in pandas 16.0.1 that causes
```
Comment (ssanderson) [inline zipline/pipeline/loaders/utils.py]: The pandas version is 0.16.1, not 16.0.1.  Is there a link to the change that we can add here so that we know when we can remove this?

File: tests/pipeline/test_events.py [hunk 3]
```diff
@@ -456,3 +460,46 @@ def make_loader(cls, events, next_value_columns, previous_value_columns):
...
+         pd.Timestamp('2013-01-31'),
+         pd.Timestamp('2013-04-04'),
+         pd.Timestamp('2013-04-21')]
+    ).reset_index(drop=True)
+
+    # Test with timezones on either side of the meridian
+    @parameterized.expand([(expected_us, 'US/Eastern', time(8, 45)),
+                           (expected_russia, 'Europe/Moscow', time(16, 45))])
+    def test_normalize_to_query_time(self, expected, tz, query_time):
+        # Order matters in pandas 18.0.2. Prior to that, using tz_convert on
```
Comment (ssanderson) [inline tests/pipeline/test_events.py]: 0.18.2

File: tests/pipeline/test_events.py [hunk 4]
```diff
@@ -456,3 +460,46 @@ def make_loader(cls, events, next_value_columns, previous_value_columns):
...
+
+    expected_us = pd.Series(
+        [pd.Timestamp('2013-01-04'),
+         pd.Timestamp('2013-01-24'),
+         pd.Timestamp('2013-02-01'),
+         pd.Timestamp('2013-04-04'),
+         pd.Timestamp('2013-04-21')]
+    ).reset_index(drop=True)
+
+    expected_russia = pd.Series(
```
Comment (ssanderson) [inline tests/pipeline/test_events.py]: Can we note here what the tz offset of Russia is?

File: tests/pipeline/test_events.py [hunk 5]
```diff
@@ -456,3 +460,46 @@ def make_loader(cls, events, next_value_columns, previous_value_columns):
...
+             pd.Timestamp('2013-04-21')]
+    combos = list(itertools.permutations(dates))
+
+    expected_us = pd.Series(
+        [pd.Timestamp('2013-01-04'),
+         pd.Timestamp('2013-01-24'),
+         pd.Timestamp('2013-02-01'),
+         pd.Timestamp('2013-04-04'),
+         pd.Timestamp('2013-04-21')]
+    ).reset_index(drop=True)
```
Comment (ssanderson) [inline tests/pipeline/test_events.py]: Why are we resetting the index here?

Comment (mtydykov) [inline tests/pipeline/test_events.py]: cruft, fixed

File: tests/pipeline/test_events.py [hunk 6]
```diff
@@ -456,3 +460,53 @@ def make_loader(cls, events, next_value_columns, previous_value_columns):
             next_value_columns,
             previous_value_columns,
         )
+
+
+class EventLoaderUtilsTestCase(ZiplineTestCase):
+    dates = pd.to_datetime([
+        '2013-01-04 3:00:00',
```
Comment (ssanderson) [inline tests/pipeline/test_events.py]: I think we should make these cases closer to the boundary conditions, e.g.:

`pd.Timestamp('2013-01-04 8:44:59', tz='US/Eastern').tz_convert('UTC')`
`pd.Timestamp('2013-01-04 8:45:00, tz='US/Eastern').tz_convert('UTC')`
